contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -186,6 +202,41 @@ contract VaultAdmin is VaultStorage {
         emit StrategyWeightsUpdated(_strategyAddresses, _weights);
     }
 
+    /**
+     * @notice Move assets from one Strategy to another
+     * @param _strategyFromAddress Address of Strategy to move assets from.
+     * @param _strategyToAddress Address of Strategy to move assets to.
+     * @param _assets Array of asset address that will be moved
+     * @param _amounts Array of amounts of each corresponding asset to move.
+     */
+    function reallocate(
+        address _strategyFromAddress,
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) external onlyGovernorOrStrategist {
+        require(
+            strategies[_strategyFromAddress].isSupported,
+            "Invalid from Strategy"
+        );
+        require(
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
+        );
+        require(_assets.length == _amounts.length, "Parameter length mismatch");
+
+        IStrategy strategyFrom = IStrategy(_strategyFromAddress);
+        IStrategy strategyTo = IStrategy(_strategyToAddress);
+
+        for (uint256 i = 0; i < _assets.length; i++) {
+            require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
```

Comment:
Should there also be a require for strategyFrom.supportsAsset(_asset[i]) ?

And is it worth having a require on strategyFrom.checkBalance(_assets[i]) <= _amounts[i] ?

Reply:
Both of those should revert so it'll save a bit of gas not to check them here. I think the to strategy check is enough 👍

Reply:
Sounds good. Thanks for checking. Let's get this in, deploy and vote for who is going to be the strategist:)
Lmk if you want me to help with the deploy stuff.